lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -335,7 +336,19 @@ export class SourceSelector extends React.Component<
 				path: selected,
 			});
 		}
-		return new sourceDestination.Http({ url: selected });
+		const splitUrl = selected.split('?');
+		const queryParams = splitUrl[splitUrl.length - 1];
```

Comment:
I think you need to check that `splitUrl` has a length greater than 1 first? :wink: 
```
foo = 'abc'.split('?')
console.log(foo[foo.length - 1])
abc
foo = 'abc?def'.split('?')
console.log(foo[foo.length - 1])
def
```

Reply:
:pray: thanks for noticing, unfortunately I changed the logic a bit and forgot to add that check.